test/Microsoft.AspNetCore.OData.Versioning.ApiExplorer.Tests/AspNet.OData/DefaultModelTypeBuilderTest.cs

```diff
@@ -408,6 +408,24 @@ public void substitute_should_resolve_types_that_reference_a_model_that_match_th
             substitutionType.Should().NotBeOfType<TypeBuilder>();
         }
 
+        [Fact]
+        public void substituted_type_should_have_renamed_with_attribute_properties_from_original_type()
+        {
+            // arrange
+            var modelBuilder = new ODataConventionModelBuilder();
+            modelBuilder.EntitySet<Contact>( "Contacts" );
+
+            var context = NewContext( modelBuilder.GetEdmModel() );
+            var originalType = typeof( Contact );
+
+            // act
+            var substitutedType = originalType.SubstituteIfNecessary( context );
+
+            // assert
+            substitutedType.Should().HaveProperty<string>(nameof( Contact.FirstName ) );
```

Comment:
This should be `"first_name"` should it not? I would expect this test to fail in it's current form.

Also, please make sure you run the **Format Document** command (assuming you're in VS) to apply all the **.editorconfig** policies/formatting.

Reply:
No, the property name is from clr, that's why I wondered that property completely 'disappeared'. Test failed prior our changes and should succeed now.

Reply:
Oh ... right, the substituted type maps the CLR property as it was, including it's attributes. So the assertion validates the property is still there. It _should_ also still have the `[DataMember]` attribute applied. Thanks for confirming.

Reply:
I added the assertion to the unit test to verify the attribute applied